tests/user/test_user_result_parser.py

```diff
@@ -523,3 +523,27 @@ def test_user_admin_custom_secret_redacted_on_error(
             f"({TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_UID_ERROR['omvs:uid']})",
             exception.exception.result,
         )
+
+    def test_user_admin_custom_secret_redacted_when_complex_characters(
+        self,
+        call_racf_mock: Mock,
+    ):
+        user_admin = UserAdmin(additional_secret_traits=["base:installation_data"])
+        call_racf_mock.side_effect = [
+            TestUserConstants.TEST_EXTRACT_USER_RESULT_BASE_ONLY_SUCCESS_XML,
+            TestUserConstants.TEST_ALTER_USER_RESULT_INST_DATA_SUCCESS_XML,
+        ]
+        result = user_admin.alter(
+            "squidwrd",
+            traits=TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_INST_DATA,
+        )
+        self.assertEqual(
+            result,
+            TestUserConstants.TEST_ALTER_USER_RESULT_SUCCESS_INST_DATA_SECRET_DICTIONARY,
+        )
+        self.assertNotIn(
+            TestUserConstants.TEST_ALTER_USER_REQUEST_TRAITS_INST_DATA[
+                "base:installation_data"
+            ],
+            result,
+        )
```

Comment:
I wonder if there should be an extract test too for the complex installation data?

Reply:
I actually am working on stabilizing this more fully in my RAS branch. There is a new test but it required new extract logic to actually work.

Reply:
I would still not consider INSTALLATION_DATA "Stable" at the end of this branch, but once this is approved, I would consider additional_secrets_redaction "stable".

Reply:
Just to confirm, redaction for extract is not within the scope of this pull request?